src/main/java/picard/analysis/CollectWgsMetricsWithNonZeroCoverage.java

```diff
@@ -179,13 +178,16 @@ public void addToMetricsFile(final MetricsFile<WgsMetrics, Integer> file,
                                      final CountingFilter dupeFilter,
                                      final CountingFilter mapqFilter,
                                      final CountingPairedFilter pairFilter) {
-            this.depthHistogram                            = getDepthHistogram();
-            final Histogram<Integer> depthHistogramNonZero = depthHistogramNonZero();
+            // calculate metrics the same way as in CollectWgsMetrics
```

Comment:
I didn't like that we stored an extra `depthHistogram` with the zero-coverage bin set to 0 in addition to the two depth histogram arrays and the base quality histogram array that `WgsMetricsCollector` already had access to. So my proposed change was: 

1. Calculate the metrics the same way as in `CollectWgsMetrics`
2. Set the zero-coverage bins in the histogram arrays in the collector to zero. Then calculate the non-zero metrics. The affected metrics (e.g. mean coverage) are calculated in `calculateDerivedFields`